src/openassetio-core/include/openassetio/EntityReference.hpp

```diff
@@ -41,16 +41,68 @@ class EntityReference final {
       : entityReferenceString_(std::move(entityReferenceString)) {}
 
   /**
-   * Compare the contents of this reference with another for equality.
+   * Compare this reference with another.
    *
-   * @param other Entity refernce to compare against.
+   * @param other Entity reference to compare against.
    *
    * @return `true` if contents are equal, `false` otherwise.
    */
-  bool operator==(const EntityReference& other) const {
+  [[nodiscard]] bool operator==(const EntityReference& other) const {
     return other.entityReferenceString_ == entityReferenceString_;
   }
 
+  /**
+   * Compare this reference with another.
+   *
+   * @param other Entity reference to compare against.
+   *
+   * @return `true` if contents are not equal, `false` otherwise.
+   */
+  [[nodiscard]] bool operator!=(const EntityReference& other) const { return !(*this == other); }
+
+  /**
+   * Compare this reference with another.
+   *
+   * @param other Entity reference to compare against.
+   *
+   * @return `true` if this reference is less than other, `false`
```

Comment:
Is it worth defining what "less than" means for an Entity reference in the public docs?

Is it equivalent to "less than" for a std::string? (lexicographical compare if I recall) I suspect so givn the implementation.

Reply:
Good shout - littered the docstrings with "lexicographically" in ca44851c